src/api/dataSource/NodeDataSource.ts

```diff
@@ -339,4 +420,34 @@ export class NodeDataSource implements NodeApi {
       return { error: { code: 500, message: 'Failed to join context.' } };
     }
   }
+  async login(loginRequest: NearRequest): ApiResponse<LoginResponse> {
+    return await this.client.post<NearRequest>(
+      `${getAppEndpointKey()}/admin-api/add-client-key`,
+      {
+        ...loginRequest,
+      },
+    );
+  }
+  async requestChallenge(): ApiResponse<NodeChallenge> {
+    return await this.client.post<NodeChallenge>(
+      `${getAppEndpointKey()}/admin-api/request-challenge`,
+      {},
+    );
+  }
+  async addRootKey(rootKeyRequest: NearRequest): ApiResponse<RootKeyResponse> {
```

Comment:
NearRequest will be updated to LoginRequest naming as this function is used by Metamask aswell